socket_test.go

```diff
@@ -260,3 +262,72 @@ func TestSocketSendSubscriptionOnConnect(t *testing.T) {
 		t.Fatalf("invalid message received: got '%s', wanted '%s'", msg.Frames[0], message)
 	}
 }
+
+type transportMock struct {
+	dialCalledCount int
+	errOnDial       bool
+	conn            net.Conn
+}
+
+func (t *transportMock) Dial(ctx context.Context, dialer transport.Dialer, addr string) (net.Conn, error) {
+	t.dialCalledCount++
+	if t.errOnDial {
+		return nil, errors.New("test error")
+	}
+	return t.conn, nil
+}
+
+func (t *transportMock) Listen(ctx context.Context, addr string) (net.Listener, error) {
+	return nil, nil
+}
+
+func (t *transportMock) Addr(ep string) (addr string, err error) {
+	return "", nil
+}
+
+func (t *transportMock) init() {
+	t.dialCalledCount = 0
+	t.errOnDial = false
+	t.conn = nil
+	zmq4.RegisterTransport("test", testTransport)
```

Comment:
that's a bit convoluted.
perhaps just register the current receiver?
```suggestion
	zmq4.RegisterTransport("test", t)
```

or register a freshly created `&transportMock{}` ?